### PR TITLE
Enh/user display name

### LINKED ISSUE #733: Display names
In V1, users had a login name and a display name:

image

We want to keep this for a variety of reasons.

- [ ] Migrate display name from v1 to v2 (I assume in V2 this should be part of the user model)
- [ ] Add a display name to the registration form
    - If left blank, populate the field with the login name
    - [ ] Ensure the display does not coincide (case-insensitively) with any existing login name or display name
    - For now, users cannot change their display name
- [ ] When posting a comment, the display name should be displayed rather than the login name

@cbunicef Are there any other instances where the display name should be displayed rather than the login name?

### PR DESCRIPTION
**Pull request by sheralim012**
fixes #733 

- added display name in user
  - added in user registration form
    - if no display name is given then use username as display name
- added in wagtail user form
- used display name instead of username in profile, comments, and chat
- updated user migration to handle display name

### CODE REVIEW

**Review by ehmadzubair — APPROVED** [review]

**Review by geoo89 — COMMENTED** [review]
When I post a comment, I get redirected to example.com

Also add the display name to the Settings > Users table as a column, and to CSV/XLSX user export.

We should make sure that the superusers we create/have created have a display name. 
I see two ways of doing this:
- patch createsuperuser and customize django migrations such that the display name is filled in during the migration (by copying in the user name) for older sites
- have an access method like display_name() in the user model that returns something like `display_name or username`.

File: iogt_users/templates/profile.html [hunk 1]
```diff
@@ -12,10 +12,10 @@
 {% block allauth_content %}
     {% if user.username %}
         <h1 class="title profile__title">
-            {% blocktranslate trimmed with username=user.username %}
-                Hey, {{ username }}
+            {% blocktranslate trimmed with display_name=user.display_name %}
+                Hey, {{ display_name }}
```
[the remark this answers is not in the document]
Comment (sheralim012) [reply iogt_users/templates/profile.html]: done

File: iogt_users/templates/profile.html [hunk 2]
```diff
@@ -12,10 +12,10 @@
...
             {% endblocktranslate %}
-            <span class="profile__icon">{{ user.username | first | upper }}</span>
+            <span class="profile__icon">{{ display_name | first | upper }}</span>
```
[the remark this answers is not in the document]
Comment (sheralim012) [reply iogt_users/templates/profile.html]: done

**Comment by sheralim012** [comment]
@geoo89 
- to resolve example.com redirection issue, update site object from settings and specify port (for local i.e., 8000 instead of 80)
- added display name to user listing and data export
- added a display_name method in user model to either return display_name or username
- used username instead of display_name in profile page

**Comment by geoo89** [comment]
When I try to migrate from an empty DB, I get the following error:

```
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/core/management/__init__.py", line 377, in execute
    django.setup()
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/apps/registry.py", line 122, in populate
    app_config.ready()
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/contrib/admin/apps.py", line 24, in ready
    self.module.autodiscover()
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/contrib/admin/__init__.py", line 24, in autodiscover
    autodiscover_modules('admin', register_to=site)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/utils/module_loading.py", line 47, in autodiscover_modules
    import_module('%s.%s' % (app_config.name, module_to_search))
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/data/idems/iogt/iogt/iogt_users/admin.py", line 4, in <module>
    from .forms import CustomUserCreationForm, CustomUserChangeForm
  File "/data/idems/iogt/iogt/iogt_users/forms.py", line 7, in <module>
    from wagtail.users.forms import UserEditForm as WagtailUserEditForm, \
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/wagtail/users/forms.py", line 181, in <module>
    class UserCreationForm(UserForm):
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/forms/models.py", line 268, in __new__
    raise FieldError(message)
django.core.exceptions.FieldError: Unknown field(s) (display_name) specified for User
```

**Comment by sheralim012** [comment]
@geoo89 fixed now

**Pull request merged**